### ISSUE TITLE
Allow mousemove event triggering or switching brush state

### ISSUE THREAD

**Issue by gena** [d3/d3-brush#78]
I'm trying to achieve something like slider control behaviour with the brush, for example, I'd like to disable handlers and only allow brush dragging, +allow clicking outside of the brush, snapping to the clicked point and then allow dragging the brush. It almost works as expected, however, when I click outside of the brush and move the brush to that location via API, the state of the brush does not change and the cursor remains crosshair. Clicking outside of the brush and immediately dragging (left or right from the brush) does not work either. Here is the notebook https://observablehq.com/d/5e5430ed51516243.

Basically, the failed behaviour is demonstrated in the following animation. And something similar to the desired behaviour, with mouse click > mouse up > move mouse (just a little to switch the brush state).

![](https://i.giphy.com/media/dB0D7NEodCxWTFLr0I/source.gif)

I think the expected behaviour can be achieved if it would be possible to trigger mousemove (or mouseclick) event within on("end") and/or on("start") handlers, but I could not find how to trigger them or switch the brush state, it looks like this is not exposed yet in the API.

**Comment by mbostock** [comment]
Does this notebook help?

https://observablehq.com/@d3/click-to-recenter-brush

**Comment by gena** [comment]
Yes, it does, perfect! Thank you!
